integration/spark/app/src/main/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -381,6 +383,19 @@ private void initializeContextFactoryIfNotInitialized(SparkConf sparkConf, Strin
   private static void initializeMetrics(OpenLineageConfig<?> openLineageConfig) {
     meterRegistry =
         MicrometerProvider.addMeterRegistryFromConfig(openLineageConfig.getMetricsConfig());
+
+    // register SimpleMeterRegistry if no other registries are present and debug facet is enabled
+    if (meterRegistry instanceof CompositeMeterRegistry
+        && ((CompositeMeterRegistry) meterRegistry).getRegistries().isEmpty()
```

Comment:
Top level registry coming from `addMeterRegistryFromConfig` will always be CompositeMeterRegistry, no need to check that.